prov/util/src/util_shm.c

```diff
@@ -68,21 +68,31 @@ static void smr_peer_addr_init(struct smr_addr *peer)
 void smr_cma_check(struct smr_region *smr, struct smr_region *peer_smr)
 {
 	struct iovec local_iov, remote_iov;
+	int remote_pid;
 	int ret;
 
-	if (peer_smr->cma_cap != SMR_CMA_CAP_NA) {
-		smr->cma_cap = peer_smr->cma_cap;
+	if (smr != peer_smr && !(peer_smr->cma_cap & SMR_PEER_CMA_CAP_NA)) {
+		smr->cma_cap = (peer_smr->cma_cap & ~0xf) + (smr->cma_cap & 0xf);
```

Comment:
I think this can be simplified with
`smr->cma_cap |= peer_smr->cma_cap & ~0xf;`